### PR TITLE
Update SOQL extension to new message structure in SOQL builder UI

### PR DESCRIPTION
**Pull request by dobladez**
### What does this PR do?

Adapts the SOQL extension to new message structure expected by the View for converting model to/from SOQL text.
Related to https://github.com/forcedotcom/soql-tooling/pull/17

Also: minor refactoring  for handling Org connection

### What issues does this PR fix or reference?
W-8038965

### CODE REVIEW

**Comment by dehru** [comment]
I changed this to use dehru/soql-to-ui-to-soql branch as the base, which i think shows the true diff.

**Review by dehru — APPROVED** [review]
This passed my QA

File: packages/salesforcedx-vscode-soql/package.json [hunk 1]
```diff
@@ -20,7 +20,7 @@
   ],
   "dependencies": {
     "@salesforce/sobject-metadata": "0.1.4",
-    "@salesforce/soql-builder-ui": "0.0.3",
+    "@salesforce/soql-builder-ui": "0.0.4",
```
Comment (dehru) [inline packages/salesforcedx-vscode-soql/package.json:23]: I published the latest and bumped this dependency.

File: packages/salesforcedx-vscode-soql/src/editor/soqlEditorInstance.ts [hunk 2]
```diff
@@ -22,18 +22,39 @@ const { OrgAuthInfo, channelService } = sfdxCoreExports;
...
+  UI_SOQL_CHANGED = 'ui_soql_changed',
   SOBJECT_METADATA_REQUEST = 'sobject_metadata_request',
   SOBJECT_METADATA_RESPONSE = 'sobject_metadata_response',
   SOBJECTS_REQUEST = 'sobjects_request',
   SOBJECTS_RESPONSE = 'sobjects_response',
-  UPDATE = 'update'
+  TEXT_SOQL_CHANGED = 'text_soql_changed'
+}
+
+// TODO: move to shared module
```
Comment (dehru) [inline packages/salesforcedx-vscode-soql/src/editor/soqlEditorInstance.ts:39]: @dobladez - Where do you think we should move them and why?

**Review by dehru — APPROVED** [review]

File: packages/salesforcedx-vscode-soql/src/editor/soqlEditorInstance.ts [hunk 3]
```diff
@@ -22,18 +22,39 @@ const { OrgAuthInfo, channelService } = sfdxCoreExports;
...
+async function getConnection(): Promise<Connection> {
+  const usernameOrAlias = await OrgAuthInfo.getDefaultUsernameOrAlias(true);
+  if (!usernameOrAlias) {
+    // TODO: NLS
+    throw new Error(
+      'No default org is set. Run "SFDX: Create a Default Scratch Org" or "SFDX: Authorize an Org" to set one.'
```
Comment (dehru) [inline packages/salesforcedx-vscode-soql/src/editor/soqlEditorInstance.ts:45]: Just noting that we have a separate work item for nls.  "W-8048923 - [soql-builder] i18n for customer-facing strings"

File: packages/salesforcedx-vscode-soql/src/editor/soqlEditorInstance.ts (hunk 2 above)
Comment (dobladez) [reply packages/salesforcedx-vscode-soql/src/editor/soqlEditorInstance.ts:39]: Hmm... I was under the impression that Org connection handling was something we needed somewhere else. 
As for `MessageType` and `SoqlEditorEvent`, I'll probably make the change (to include them from `sql-builder-ui`) as part of my next work item).

File: packages/salesforcedx-vscode-soql/src/editor/soqlEditorInstance.ts [hunk 4]
```diff
@@ -22,18 +22,39 @@ const { OrgAuthInfo, channelService } = sfdxCoreExports;
...
   SOBJECT_METADATA_RESPONSE = 'sobject_metadata_response',
   SOBJECTS_REQUEST = 'sobjects_request',
   SOBJECTS_RESPONSE = 'sobjects_response',
-  UPDATE = 'update'
+  TEXT_SOQL_CHANGED = 'text_soql_changed'
+}
+
+// TODO: move to shared module
+async function getConnection(): Promise<Connection> {
+  const usernameOrAlias = await OrgAuthInfo.getDefaultUsernameOrAlias(true);
```
Comment (brpowell) [inline packages/salesforcedx-vscode-soql/src/editor/soqlEditorInstance.ts:41]: Passing true into this call will prompt the same message in the error below as a notification and in the output window. Do you still want to throw this error?

File: packages/salesforcedx-vscode-soql/src/editor/soqlEditorInstance.ts (hunk 2 above)
Comment (brpowell) [reply packages/salesforcedx-vscode-soql/src/editor/soqlEditorInstance.ts:39]: You could update `getConnection` in the OrgAuthInfo util to accept either a `usernameOrAlias` or `undefined`. If it's undefined it'll just use the default one like you're doing here.

File: packages/salesforcedx-vscode-soql/src/editor/soqlEditorInstance.ts [hunk 5]
```diff
@@ -22,18 +22,39 @@ const { OrgAuthInfo, channelService } = sfdxCoreExports;
...
+  }
+  return await OrgAuthInfo.getConnection(usernameOrAlias);
+}
+
+async function withSFConnection(f: (conn: Connection) => void): Promise<void> {
+  try {
+    const conn = await getConnection();
+    f(conn);
+  } catch (e) {
+    channelService.appendLine(e);
```
Comment (xyc) [inline packages/salesforcedx-vscode-soql/src/editor/soqlEditorInstance.ts:56]: Ideally we'd wish to have some telemetry here.

Comment (dehru) [reply packages/salesforcedx-vscode-soql/src/editor/soqlEditorInstance.ts:56]: Yes.  We are now just working aggressively towards an MVP and plan on swinging back through for telemetry and localization.  It's been prioritized like that for us.  

Future work item is:
W-8086238 - [soql-builder] Add telemetry to get insight into soql builder usage / errors

Comment (dobladez) [reply packages/salesforcedx-vscode-soql/src/editor/soqlEditorInstance.ts:56]: If I understood correctly, if we do what @brpowell  suggests, we might not event need the `catch`

Comment (dobladez) [reply packages/salesforcedx-vscode-soql/src/editor/soqlEditorInstance.ts:44]: I've made the change suggested by @brpowell, but we might still need that catch anyway

File: packages/salesforcedx-vscode-core/test/vscode-integration/util/authInfo.test.ts [hunk 6]
```diff
@@ -79,4 +80,24 @@ describe('OrgAuthInfo', () => {
       infoMessageStub.restore();
     });
   });
+
+  describe('getConnection', () => {
+    const username = 'user@test.test';
+    const alias = 'TestOrg';
+
+    it('with argument', async () => {
```
Comment (brpowell) [inline packages/salesforcedx-vscode-core/test/vscode-integration/util/authInfo.test.ts:88]: test case conventions include "should" e.g.

`it('should return connection associated with given username')`

`it('should return connection with the default username if one is not provided')`

**Review by brpowell — APPROVED** [review]
Just the one comment about test case names.

File: packages/salesforcedx-vscode-core/test/vscode-integration/util/authInfo.test.ts (hunk 6 above)
Comment (dobladez) [reply packages/salesforcedx-vscode-core/test/vscode-integration/util/authInfo.test.ts:88]: OK. I'm changing them on my next PR. Thanks!

**Pull request merged**